Src/java/engine/src/main/java/org/opencds/cqf/cql/engine/execution/CqlEngine.java

```diff
@@ -315,4 +372,1416 @@ private Set<String> getExpressionSet(Library library) {
 
         return expressionNames;
     }
+
+    public void processException(Exception e, Element element) {
+        if (e instanceof CqlException) {
+            CqlException ce = (CqlException) e;
+            if (ce.getSourceLocator() == null) {
+                ce.setSourceLocator(SourceLocator.fromNode(element, null));
+                DebugAction action = state.shouldDebug(ce);
+                if (action != DebugAction.NONE) {
+                    state.logDebugError(ce);
+                }
+            }
+            throw (RuntimeException) (e);
+        } else {
+            CqlException ce = new CqlException(e, SourceLocator.fromNode(element, null));
+            DebugAction action = state.shouldDebug(ce);
+            if (action != DebugAction.NONE) {
+                state.logDebugError(ce);
+            }
+            throw (RuntimeException) ce;
+        }
+    }
+
+    @Override
+    public Object visitExpressionDef(ExpressionDef expressionDef, State state) {
+
+        //logger.info(String.format("expression def visit: %s", expressionDef.getName()));
+        return ExpressionDefEvaluator.internalEvaluate(expressionDef, state, this);
+    }
+
+    public Object validateOperand(Object operand) {
```

Comment:
It's unclear to me why we need this. Shouldn't `ExpressionDefs` be visited already when resolving the operand?